File: Linked List/java/delete_node.java
public class Linkedlist {
    Node head;

    static class Node
    {
        int data;
        Node link;

        Node(int num){
            data = num;
            link = null;
        }
    }

    //insert a node at the end of linked list
    public void append(int num){
        Node temp = new Node(num);
        if(head == null){
            head = temp;
            return;
        }
        Node temp1 = head;
        while (temp1.link!=null){
            temp1 = temp1.link;
        }

        temp1.link = temp;
    }
  
    //delete a node
    public void delete(int loc){
        Node temp = head;
        if(loc==1){
            head = temp.link;
            return;
        }
        loc -=2;
        while (loc>0){
            temp = temp.link;
            loc--;
        }

        Node temp1 = temp.link;
        temp.link = temp1.link;
    }
  
    //print linked list
    public void print(){
        Node temp = head;
        while (temp!=null){
            System.out.print(temp.data + " ");
            temp = temp.link;
        }
    }
  public static void main(String[] args){
        //Scanner scan = new Scanner(System.in);
        Linkedlist list = new Linkedlist();
        list.append(1);
        list.append(2);
        list.append(3);
        list.append(4);
        list.append(5);
    
        list.delete(3);

        list.print();
    }
}

File: Linked List/delete_node.cpp
#include <iostream>
using namespace std;

class Node 
{
public:
    int data;
    Node* link;

    Node(int num){
        data = num;
        link = NULL;
    } 
};

Node* head; //pointer to head node

// end
void insert_end(int num){
    Node* temp = new Node(num);

    if(head == NULL){
        head = temp;
        return;
    }

    Node* temp1 = head;
    while (temp1->link != NULL)
    {
        temp1 = temp1->link;
    }

    temp1->link = temp;   
}


// delete node

void remove(int loc){
    Node *temp1 = head;

    if(loc == 1){
       head = temp1->link;
       delete temp1;
       return;
    }
    loc -= 2;

    while (loc--)
    {
       temp1 = temp1->link; 
    }
    Node *temp = temp1->link;
    temp1->link = temp->link;
    delete temp;
}

// print linked list

void print(){
    Node *temp = head;
    while (temp!=NULL)
    {
        cout<<temp->data<<" ";
        temp = temp->link;
    }
}

// main

int main(){
    int list_size;
    cin>>list_size;
    head = NULL;

    while (list_size--)
    {
        int num;
        cin>>num;
        insert_end(num);
    }

    remove(3);

    print();
}
